src/lib/common/traits.hpp

```diff
@@ -1077,6 +1077,12 @@ inline T const& escape(T const& x) {
 }
 //! @}
 
+/**
+ * @brief Declares that the type @param T is usable as an output stream. 
+ * Enabled by default for all subtypes of std::ostream.
+ */
+template <typename T>
+struct is_ostream : public std::is_base_of<std::ostream, T> {};
```

Comment:
add `if_ostream` as a `using` of `std::enable_if` of a `is_ostream` for convenience

Reply:
Also, you should add a test for this feature. By doing so, you would have noticed that you are missing a `#include <ostream>` directive in this file